applications/nrf_desktop/src/modules/swift_pair.c

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <bluetooth/adv_prov/swift_pair.h>
+
+#define MODULE swift_pair_app
+#include <caf/events/ble_common_event.h>
+#include "ble_dongle_peer_event.h"
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(MODULE, CONFIG_DESKTOP_SWIFT_PAIR_LOG_LEVEL);
+
+#define DEFAULT_APP_ID	0
+#define INVALID_APP_ID	UINT8_MAX
+
+static uint8_t dongle_app_id = INVALID_APP_ID;
+
+
+static void update_swift_pair_payload(uint8_t selected_app_id)
+{
+	bool enable;
+
+	if (selected_app_id == dongle_app_id) {
+		enable = IS_ENABLED(CONFIG_DESKTOP_SWIFT_PAIR_ADV_DONGLE_PEER);
+	} else {
+		enable = IS_ENABLED(CONFIG_DESKTOP_SWIFT_PAIR_ADV_OTHER_BLE_PEER);
+	}
+
+	bt_le_adv_prov_swift_pair_enable(enable);
+	LOG_DBG("Swift Pair payload %sabled", enable ? "en" : "dis");
+}
+
+static bool handle_ble_dongle_peer_event(const struct ble_dongle_peer_event *event)
+{
+	__ASSERT_NO_MSG(event->bt_app_id != INVALID_APP_ID);
+
+	dongle_app_id = event->bt_app_id;
+	update_swift_pair_payload(DEFAULT_APP_ID);
```

Comment:
Is it guaranteed that the nRF Desktop will start with the default application identity? Shouldn't we rely on the peer operation logic here?

Reply:
In general, we actually rely on the peer operation logic here.
Still, in case peer operation logic would not be enabled in the application, we may not receive the event. In that case, we could assume that we use the default application identity (to be consistent with approach used in the CAF BLE adv module).